python/pyarrow/pandas_compat.py

```diff
@@ -822,8 +822,12 @@ def _get_extension_dtypes(table, columns_metadata, types_mapper=None):
 
     # infer the extension columns from the pandas metadata
     for col_meta in columns_metadata:
-        name = col_meta['name']
+        if col_meta['name']:
+            name = col_meta['name']
+        else:
+            name = col_meta['field_name']
```

Comment:
If we only use `col_meta["field_name"]` we get KeyErrors for the missing `filed_name` in the parquet tests. Something to do with fastparquet (0.3.2)? 

https://github.com/apache/arrow/blob/ad073b7c0fec80ce88aaf1e7d6a78104711952f2/python/pyarrow/tests/test_pandas.py#L4258-L4262

Reply:
Ah, yes, that's for compatibility with old metadata where the "field_name" can be missing. Maybe instead of this if/else, you could then also do 

```
try:
    name = col_meta["field_name"]
except KeyError:
    name = col_meta["name"]
```

so it is clearer that this is a fallback.